core/src/main/scala/dagr/core/execsystem/TaskExecutionRunner.scala

```diff
@@ -270,6 +269,9 @@ private[core] class TaskExecutionRunner extends TaskExecutionRunnerApi with Lazy
       }
   }
 
+  /** Join on all the running processes until they are finished or the timeout reached. */
+  override def joinAll(millis: Long): Unit = this.processes.values.foreach(_.join(millis))
```

Comment:
this may join on each for `millis` rather until the timeout is reached.  Update the doc or the desired behavior.